src/services/reports.js

```diff
@@ -0,0 +1,359 @@
+
+						// this filter is applied for the below condition
+						// SM mentee_visibility_policy (in case of mentee list) or external_mentor_visibility policy (in case of mentor list) = ALL
+						//  and CASE 1 (mentee list) : Mentees is related to the SM org but external_mentee_visibility is CURRENT (exclude these mentees)
+						//  CASE 2 : (session list) : Sessions is related to the SM org but session_visibility is CURRENT (exclude these sessions)
+						//  CASE 3 : (mentor list) : Mentors is related to SM Org but mentor_visibility set to CURRENT  (exclude these mentors)
+						const additionalFilter =
+							filterType == common.MENTEE_ROLE
+								? {
+										mentee_visibility_policy: {
+											//1
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+								: filterType == common.SESSION
+								? {
+										session_visibility_policy: {
+											//2
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+								: {
+										mentor_visibility_policy: {
+											//3
+											[Op.ne]: 'CURRENT',
+										},
+								  }
+						const organizationExtension = await organisationExtensionQueries.findAll(
+							{
+								[Op.or]: [
+									filterQuery,
+									{
+										[Op.and]: [
+											{
+												organization_id: {
+													[Op.in]: [...relatedOrgs],
+												},
+											},
+											additionalFilter,
+										],
+									},
+								],
+							},
+							{
+								attributes: ['organization_id'],
+							}
+						)
+						organizationIds.push(orgExtension.organization_id)
+						if (organizationExtension) {
+							const organizationIdsFromOrgExtension = organizationExtension.map(
+								(orgExt) => orgExt.organization_id
+							)
+							organizationIds.push(...organizationIdsFromOrgExtension)
+						}
+					}
+				}
+			}
+
+			return {
+				success: true,
+				result: organizationIds,
+			}
+		} catch (error) {
+			return {
+				success: false,
+				message: error.message,
+			}
+		}
+	}
+
+	static async getEntityTypeWithEntitiesBasedOnOrg(organization_ids, entity_types, defaultOrgId = '', modelName) {
+		try {
+			let filter = {
+				status: common.ACTIVE_STATUS,
+				allow_filtering: true,
+				has_entities: true,
+				organization_id: {
+					[Op.in]: defaultOrgId ? [...organization_ids, defaultOrgId] : organization_ids,
+				},
+				report_filter: true,
+			}
+
+			let entityTypes = []
+			if (entity_types) {
+				entityTypes = entity_types.split(',')
+				filter.value = {
+					[Op.in]: entityTypes,
+				}
+			}
+
+			if (modelName) {
+				filter.model_names = { [Op.contains]: [modelName] }
+			}
+			//fetch entity types and entities
+			let entityTypesWithEntities = await entityTypeQueries.findUserEntityTypesAndEntities(filter)
+
+			return {
+				success: true,
+				result: entityTypesWithEntities,
+			}
+		} catch (error) {
+			return {
+				success: false,
+				message: error.message,
+			}
+		}
+	}
+}
+
+function convertEntitiesForFilter(entityTypes) {
```

Comment:
@sumanvpacewisdom as the same code used in meentes.js as well, move this utils and make changes in meentes.js as well